django/http/cookie.py

```diff
@@ -91,16 +91,17 @@ def _BaseCookie__set(self, key, real_value, coded_value):
 def parse_cookie(cookie):
     if cookie == '':
         return {}
-    if not isinstance(cookie, http_cookies.BaseCookie):
-        try:
-            c = SimpleCookie()
-            c.load(cookie)
-        except http_cookies.CookieError:
-            # Invalid cookie
-            return {}
-    else:
-        c = cookie
+    if hasattr(cookie, 'keys'):  # SimpleCookie instance
+        return {k: cookie.get(k).value for k in cookie.keys()}
+    cookie = force_str(cookie)  # for backward compatibility on Python 2
```

Comment:
Ahh. I see the tests pass without it. Maybe we can add it bad in if anyone complains.